packages/node_modules/pouchdb-adapter-indexeddb/src/index.js

```diff
@@ -18,152 +17,117 @@ import purge from './purge';
-        if (typeof last === 'function') {
-          last(err);
-        } else {
-          console.error(err);
-        }
-      });
-    };
-  };
-  // the promise version of $
-  var $p = function (fun) {
-    return function () {
-      var args = Array.prototype.slice.call(arguments);
-
-      return setup(openDatabases, api, dbOpts).then(function (res) {
-        metadata = res.metadata;
-        args.unshift(res.idb);
-
-        return fun.apply(api, args);
-      });
-    };
-  };
+  const $ = (fun) => (...args) => 
+  $p(fun)(...args).catch((err) => 
+    typeof args.slice(-1) === 'function' ? args.slice(-1)(err)
+     : console.error(err);
+ 
   // Wrapper that gives you a safe transaction handle. It's important to use
   // this instead of opening your own transaction from a db handle got from $,
   // because in the time between getting the db handle and opening the
   // transaction it may have been invalidated by index changes.
-  var $t = function (fun, stores, mode) {
-    stores = stores || [DOC_STORE];
-    mode = mode || 'readonly';
-
-    return function () {
-      var args = Array.prototype.slice.call(arguments);
-      var txn = {};
-      setup(openDatabases, api, dbOpts).then(function (res) {
+  const $t = (fun, stores = [DOC_STORE], mode = 'readonly') => {
+    return (...args) => {
+      const txn = {};
+      setup(openDatabases, api, dbOpts).then(res => {
         metadata = res.metadata;
         txn.txn = res.idb.transaction(stores, mode);
-      }).catch(function (err) {
+      },err => {
         console.error('Failed to establish transaction safely');
         console.error(err);
         txn.error = err;
-      }).then(function () {
-        args.unshift(txn);
-        fun.apply(api, args);
-      });
+      }).then(() =>
+        fun.apply(api, [txn,...args]);
+      );
     };
   };
 
-  api._openTransactionSafely = function (stores, mode, callback) {
-    $t(function (txn, callback) {
+  api["_openTransactionSafely"] = (stores, mode, callback) => {
+    $t((txn, callback) => {
       callback(txn.error, txn.txn);
     }, stores, mode)(callback);
   };
 
-  api._remote = false;
-  api.type = function () { return ADAPTER_NAME; };
+  api["_remote"] = false;
+  api["type"] = () => ADAPTER_NAME;
 
-  api._id = $(function (_, cb) {
+  api["_id"] = $((_, cb) => {
     cb(null, metadata.db_uuid);
   });
 
-  api._info = $(function (_, cb) {
-    return info(metadata, cb);
-  });
+  api["_info"] = $((_, cb) => info(metadata, cb));
 
-  api._get = $t(get);
+  api["_get"] = $t(get);
 
-  api._bulkDocs = $(function (_, req, opts, callback) {
+  api["_bulkDocs"] = $((_, req, opts, callback) => {
     bulkDocs(api, req, opts, metadata, dbOpts, idbChanges, callback);
   });
 
-  api._allDocs = $t(function (txn, opts, cb) {
+  api["_allDocs"] = $t((txn, opts, cb) => {
     allDocs(txn, metadata, opts, cb);
   });
 
-  api._getAttachment = $t(getAttachment);
+  api["_getAttachment"] = $t(getAttachment);
```

Comment:
I'm not convinced this style is an improvement - why do you prefer this?

Reply:
It is only a readability change to make it more easy to see where the code is that matters most parts of the code are relativ useless adapters as you maybe know this change allowed me to do faster code analyses as you used inheritance all over. If you want to keep this project alive which is far far far behind the official couchdb docs. then Style should not be the part to worry about...

but if you have a genuine interest to know if this style is a regression in terms of performance no it is not! https://chromium.googlesource.com/v8/v8.git/+/refs/heads/main/AUTHORS#120 i relativ sure know what i say at last i am familar with the v8 assembler and i am the creator of B8G the v8 Compiler feedback interface.